src/platform/packages/shared/kbn-rrule/rrule.ts

```diff
@@ -74,19 +32,15 @@ type AllResult = Date[] & {
 };
 
 const ALL_LIMIT = 10000;
+const TIMEOUT_LIMIT = 100000;
 
 export class RRule {
   private options: Options;
   constructor(options: ConstructorOptions) {
-    this.options = options as Options;
-    if (isNaN(options.dtstart.getTime())) {
-      throw new Error('Cannot create RRule: dtstart is an invalid date');
-    }
-    if (options.until && isNaN(options.until.getTime())) {
-      throw new Error('Cannot create RRule: until is an invalid date');
-    }
+    this.options = sanitizeOptions(options as Options);
     if (typeof options.wkst === 'string') {
```

Comment:
I think this `if` block could/should be moved to `sanitizeOptions`. it's exactly the kind of thing that is done there so there is no point in splitting the logic.

Reply:
also, I checked usages of RRule and a string is never passed here. We should probably change the type of `wkst` to `wkst?: Weekday;` and never allow `null`.

Reply:
> I think this if block could/should be moved to sanitizeOptions. it's exactly the kind of thing that is done there so there is no point in splitting the logic.

agree

Reply:
Ok, I just found that the `constructor`'s actual type of `wkst` is different.

`type ConstructorOptions = Omit<Options, 'byweekday' | 'wkst'> & {
  byweekday?: Array<string | number> | null;
  wkst?: Weekday | WeekdayStr | number | null;
};
`

I am resolving this.